pkg/cloudprovider/providers/vsphere/vclib/diskmanagers/vmdm.go

```diff
@@ -44,16 +68,16 @@ func (vmdisk vmDiskManager) Create(ctx context.Context, datastore *vclib.Datasto
 		}
 	} else {
 		glog.Errorf("volumeOptions.StoragePolicyID or volumeOptions.VSANStorageProfileData is needed for VM based Volume Provisioning")
-		return fmt.Errorf("volumeOptions.StoragePolicyID or volumeOptions.VSANStorageProfileData is not set")
+		return fmt.Errorf("volumeOptions.StoragePolicyID and volumeOptions.VSANStorageProfileData is not set")
```

Comment:
Error message is not clear. You can rephrase like this
volumeOptions.StoragePolicyID and volumeOptions.VSANStorageProfileData is not set -> Both volumeOptions.StoragePolicyID and volumeOptions.VSANStorageProfileData are not set. One of them should be set.